src/main/java/com/tfgm/models/TramStop.java

```
@@ -1,82 +1,231 @@
 package com.tfgm.models;
 
 import java.util.ArrayList;
-
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * The main TramStop entity. This entity represents the combination of a tram stop and direction.
+ *
+ * @author aelina
+ */
 public class TramStop {
 
-  private String stopName, direction, line;
-  private int id;
-  private ArrayList<TramTrip> tramTrips = new ArrayList<>();
-
-  public TramStop(String stopName, String line, String direction, int id) {
+  /** The offical name of the tram stop. */
+  private final String stopName;
+
+  /**
+   * The direction a tram stop is going. Incoming is towards the centre of manchester, Outgoing is
+   * away.
+   */
+  private final String direction;
+
+  /** The overall line the stop is on. */
+  private final String line;
+
+  /**
+   * An arraylist of the previous updates this TramStop has had. Generally in the format
+   * "YYYYMMDDHHMM{DestinationName}."
+   */
+  private final ArrayList<String> lastUpdated = new ArrayList<>();
+
+  /** A representation of Trams currently at this Station going in a direction. */
+  private final Queue<Tram> tramQueue = new LinkedList<>();
+
+  /**
+   * The next nodes in the graph. Please see {@link com.tfgm.models.TramStopContainer} for more
+   * details.
+   */
+  private TramStopContainer[] nextStops;
+  /**
+   * The next previous nodes in the graph. Please see {@link com.tfgm.models.TramStopContainer} for
+   * more details.
+   */
+  private TramStopContainer[] prevStops;
+
+  /**
+   * Constructor for class TramStop.
+   *
+   * @param stopName Name of the stop.
+   * @param direction Direction of stop.
+   * @param line Line of the stop.
+   */
+  public TramStop(String stopName, String direction, String line) {
     this.stopName = stopName;
-    this.line = line;
     this.direction = direction;
-    this.id = id;
+    this.line = line;
```

Are these values really nullable.  Are empty and whitespace only strings allowed for instance?

---

Unsure what you're asking. Whitespace and empty would be allowed they just would never come up

---

It normally to have checks on arguments for models and boundary layers validating or asserting constraints as appropriate.  This enforces method contracts and is an implementation of the fail fast principle rather than allowing errors to propagate in confusing and difficult to track down ways.